datastore/src/main/java/com/google/datastore/snippets/Concepts.java

```diff
@@ -0,0 +1,1058 @@
+        .kind("Task")
+        .filter(CompositeFilter.and(PropertyFilter.eq("priority", 4),
+            PropertyFilter.gt("created", startDate), PropertyFilter.lt("created", endDate)))
+        .build();
+    // [END equal_and_inequality_range]
+    assertValidQuery(query);
+  }
+
+  @Test
+  public void testInequalitySort() {
+    setUpQueryTests();
+    // [START inequality_sort]
+    Query<Entity> query = Query.entityQueryBuilder()
+        .kind("Task")
+        .filter(PropertyFilter.gt("priority", 3))
+        .orderBy(OrderBy.asc("priority"), OrderBy.asc("created"))
+        .build();
+    // [END inequality_sort]
+    assertValidQuery(query);
+  }
+
+  @Test
+  public void testInequalitySortInvalidNotSame() {
+    // [START inequality_sort_invalid_not_same]
+    Query<Entity> query = Query.entityQueryBuilder()
+        .kind("Task")
+        .filter(PropertyFilter.gt("priority", 3))
+        .orderBy(OrderBy.asc("created"))
+        .build();
+    // [END inequality_sort_invalid_not_same]
+    assertInvalidQuery(query);
+  }
+
+  @Test
+  public void testInequalitySortInvalidNotFirst() {
+    // [START inequality_sort_invalid_not_first]
+    Query<Entity> query = Query.entityQueryBuilder()
+        .kind("Task")
+        .filter(PropertyFilter.gt("priority", 3))
+        .orderBy(OrderBy.asc("created"), OrderBy.asc("priority"))
+        .build();
+    // [END inequality_sort_invalid_not_first]
+    assertInvalidQuery(query);
+  }
+
+  @Test
+  public void testLimit() {
+    setUpQueryTests();
+    // [START limit]
+    Query<Entity> query = Query.entityQueryBuilder().kind("Task").limit(5).build();
+    // [END limit]
+    assertValidQuery(query);
+  }
+
+  @Test
+  public void testCursorPaging() {
+    setUpQueryTests();
+    datastore.put(testEntity);
+    Cursor nextPageCursor = cursorPaging(1, null);
+    assertNotNull(nextPageCursor);
+    nextPageCursor = cursorPaging(1, nextPageCursor);
+    assertNotNull(nextPageCursor);
+  }
+
+  private Cursor cursorPaging(int pageSize, Cursor pageCursor) {
+    // [START cursor_paging]
+    EntityQuery.Builder queryBuilder = Query.entityQueryBuilder().kind("Task").limit(pageSize);
+    if (pageCursor != null) {
+      queryBuilder.startCursor(pageCursor);
+    }
+    QueryResults<Entity> tasks = datastore.run(queryBuilder.build());
+    Entity task = null;
+    while (tasks.hasNext()) {
+      task = tasks.next();
+      // do something with the task
+    }
+    Cursor nextPageCursor = null;
+    if (task != null && tasks.cursorAfter() != null) {
+      nextPageCursor = tasks.cursorAfter();
+      // Call nextPageCursor.toUrlSafe() if you want an encoded cursor that can be used as part of a
+      // URL.
+    }
+    // [END cursor_paging]
+    return nextPageCursor;
+  }
+
+  @Test
+  public void testEventualConsistentQuery() {
+    setUpQueryTests();
+    // [START eventual_consistent_query]
+    Query<Entity> query = Query.entityQueryBuilder()
```

Comment:
Yes, it is supported in both via `RunQueryRequest.read_options`:

    runQueryRequestBuilder = RunQueryRequest.newBuilder()
    runQueryRequestBuilder.getReadOptionsBuilder().setReadConsistency(ReadConsistency.EVENTUAL);